src/objects.rs

```diff
@@ -144,34 +145,40 @@ impl Object {
         &self,
         commands: &'b mut Commands<'a>,
         texture_atlas: Option<&Handle<TextureAtlas>>,
+        atlas_sprite_index: Option<u32>,
         map: &tiled::Map,
         map_handle: Handle<Map>,
         tile_map_transform: &Transform,
         debug_config: &DebugConfig,
     ) -> EntityCommands<'a, 'b> {
         let mut new_entity_commands = if let Some(texture_atlas) = texture_atlas {
-            let sprite_index = self.sprite_index.expect("missing sprite index");
+            let atlas_sprite_index = atlas_sprite_index.expect("missing atlas sprite index");
             let tileset_gid = self.tileset_gid.expect("missing tileset");
 
             // fetch tile for this object if it exists
-            let object_tile_size = map
+            let tileset = map
                 .tilesets
                 .iter()
-                .find(|ts| ts.first_gid == tileset_gid)
+                .find(|ts| ts.first_gid == tileset_gid);
+            let object_tile_size = tileset
                 .map(|ts| Vec2::new(ts.tile_width as f32, ts.tile_height as f32));
             // object dimensions
             let dims = self.dimensions();
             // use object dimensions and tile size to determine extra scale to apply for tile objects
-            let tile_scale = if let (Some(dims), Some(size)) = (dims, object_tile_size) {
-                Some((dims / size).extend(1.0))
+            let tile_scale = if tileset.map(|tileset| tileset.images.len() > 0) == Some(true) {
+                if let (Some(dims), Some(size)) = (dims, object_tile_size) {
+                    Some((dims / size).extend(1.0))
+                } else {
+                    None
+                }
             } else {
-                None
+                Some(Vec3::ONE)
```

Comment:
This provides the correct positioning and scaling for individual image tiles, since their sizing has nothing to do with the tileset's grid.